### PR TITLE
[IN-331][IN-284][OSF] Withdrawal-related emails to moderators and contributors.

### PR DESCRIPTION
**Pull request by adlius**
## Purpose

This PR adds two types of notification emails sent to users:
- When a contributor submits a withdrawal request, emails are sent to **moderators**.
- When a preprint is eventually withdrawn, emails are sent to **contributors**.

## Changes

Two major changes are made accordingly:
- `notify_withdraw` hook for `ReviewsMachine` is implemented to send email when a preprint is withdrawn.
    - A newly added email template `PREPRINT_WITHDRAWAL_REQUEST_GRANTED` is added.

- `notify_submit` hook for `PreprintRequestMachine` is implemented to send email when a withdrawal request is submitted.

## QA Notes

For emails sent to **contributors**, there are two cases:

- When a contributor submits a withdrawal request and a moderator approves such request
    - An email should be sent to all contributors, including the request submitter. **However the language is different for the request submitter.**

- When a moderator withdraws a preprint
    - An email should be sent to all contributors. The language should say that a moderator performs such withdrawal.

For emails sent to **moderators**, they should be in the form of a digest email.

## Ticket

https://openscience.atlassian.net/browse/IN-331
https://openscience.atlassian.net/browse/IN-284

### CODE REVIEW

File: osf/utils/machines.py [hunk 1]
```diff
@@ -246,23 +262,22 @@ def save_changes(self, ev):
         elif ev.event.name == DefaultTriggers.SUBMIT.value:
             # If the provider is pre-moderated and target has not been through moderation, auto approve withdrawal
             if self.auto_approval_allowed():
-                self.machineable.target.run_withdraw(user=self.machineable.creator, comment=self.action.comment)
+                self.machineable.run_accept(user=self.machineable.creator, comment=self.action.comment)
```
Comment (mfraezz) [inline osf/utils/machines.py]: 👍

File: website/reviews/listeners.py [hunk 2]
```diff
@@ -86,3 +86,47 @@ def reviews_submit_notification_moderators(self, timestamp, context):
...
+    # Set url for profile image of the submitter
+    context['profile_image_url'] = get_profile_image_url(context['requester'])
+    # Set submission url
+    context['reviews_submission_url'] = '{}reviews/preprints/{}/{}'.format(settings.DOMAIN,
+                                                                           preprint.provider._id,
+                                                                           preprint._id)
+    # Store emails to be sent to subscribers instantly (at a 5 min interval)
+    emails.store_emails(provider_subscription.email_transactional.all().values_list('guids___id', flat=True),
+                        'email_transactional',
+                        'new_pending_submissions',
```
Comment (mfraezz) [inline website/reviews/listeners.py]: Initially, I was concerned that this was the wrong template to be using here, but looking into it a bit, it seems almost configurable enough to be useable.

However, there's an issue with using it that I don't see addressed yet -- the `person` `span` isn't guaranteed to return just one person:
```
<span class="person" style="font-weight: bold;">${', '.join(reviewable.node.contributors.values_list('fullname', flat=True))}</span>
```
This could lead to email text along the lines of
```
Alice Allison, Bob Bobson, Carol Carolson [etc.] has requested withdrawal of [...]
```
which is definitely incorrect. Recommended fix: additional `context` var, conditional in the template.

File: website/mails/mails.py [hunk 3]
```diff
@@ -421,3 +421,18 @@ def get_english_article(word):
     'crossref_doi_error',
     subject='There was an error creating a DOI for preprint(s). batch_id: ${batch_id}'
 )
+
+CROSSREF_CSV = Mail(
+    'crossref_csv',
+    subject='[auto] Here is a CSV of DOIs related to crossref: ${csv_type}'
+)
+
+PREPRINT_DOI_CHANGE = Mail(
```
Comment (mfraezz) [inline website/mails/mails.py]: This and the above template appear unrelated

**Comment by adlius** [comment]
Removed unrelated email templates. TCI build is failing due to some upstream problem related to citation formats.
